### PR TITLE
Added critical section support needed in esp32-hal & added license and readme info

### PR DESCRIPTION
**Pull request by arjanmels**
- Cleanup and additional info for cargo.toml
- Added interrupt enable/disable and critical section
- Added README.md
- Added license and readme info

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,27 +1,19 @@
 [package]
 name = "xtensa-lx6-rt"
-version = "0.1.0"
-authors = ["Scott Mabin <scott@mabez.dev>"]
+version = "0.2.0"
+description = "Low level access for xtensa lx6 processors"
+categories = ["embedded", "hardware-support", "no-std"]
+keywords = ["xtensa", "lx6", "register", "peripheral"]
+license = "MIT OR APache-2.0"
```
Comment (jeandudey) [inline Cargo.toml]:
```suggestion
license = "MIT OR Apache-2.0"
```

File: src/macros.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+///
+/// This macro returns a value with type `Option<&'static mut $ty>`. `Some($expr)` will be returned
+/// the first time the macro is executed; further calls will return `None`. To avoid `unwrap`ping a
+/// `None` variant the caller must ensure that the macro is called from a function that's executed
+/// at most once in the whole lifetime of the program.
+///
+/// # Example
+///
+/// ``` no_run
+/// use cortex_m::singleton;
```
Comment (jeandudey) [inline src/macros.rs]:
```suggestion
/// use xtensa_lx6_rt::singleton;
```

Comment (arjanmels) [reply src/macros.rs]: Nicely caught, thanks!

**Comment by MabezDev** [comment]
LGTM, thanks!

**Pull request merged**